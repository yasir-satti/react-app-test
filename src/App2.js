import React from "react";
// import ReactDOM from "react-dom";
import "./index.css";

// 2. example app:
// https://www.freecodecamp.org/news/react-examples-reactjs/

function Welcome(props) {
  return <h1>Hello, {props.name}</h1>;
}

function App2() {
  return <Welcome name="Mr. Faisal Arkan" />;
  // return ( <div>Faisal Arkam</div> )
  // return ( <body><div><p>Faisal Akram</p></div></body> )
}

// ReactDOM.render(element, document.getElementById("root"));

export default App2;
