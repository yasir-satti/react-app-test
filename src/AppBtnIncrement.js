import React, { useState } from 'react';
import './App.css';

function AppBtnIncrement() {
const [number, setNumber] = useState(0)
const onClick = () => {
    setTimeout(
        () => {
            const num = number + 1;
            setNumber(num)
        },
        1000
    )
}
return (
<div className="App">
    <span>
        {number}
    </span>
    <button onClick={onClick}>
         Button
    </button>
</div >
);
}

export default AppBtnIncrement;