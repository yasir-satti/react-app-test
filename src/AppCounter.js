import { useState } from 'react';
import App2 from "./App2"
import Counter from "./Counter"
import TestHook from './TestHook';

const AppCounter = () => {
    const [state, setState] = useState("Some Text")
    const [name, setName] = useState("Moe")

    const changeName = () => {
      setName("Steve")
    }

    return (
      <div className="App">
        <App2 />
        <h1> Counter </h1>
        <Counter />
        <h1> Basic Hook useState </h1>
        <TestHook name={name} changeName={changeName}/>
       </div>
    )
}

export default AppCounter;