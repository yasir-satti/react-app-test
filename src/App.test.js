import React from "react";
import ReactDOM from "react-dom";
// import { render, screen } from '@testing-library/react';
import App from "./App";
import App2 from "./App2";
import Counter from "./Counter";
import TestHook from "./TestHook.js";
import AppBtnIncrement from "./AppBtnIncrement";
import { render, fireEvent, cleanup, waitFor } from "@testing-library/react";

import TestRenderer from "react-test-renderer";
import ShallowRenderer from "react-test-renderer/shallow";

// import Enzyme, { shallow, render, mount } from 'enzyme';
import toJson from "enzyme-to-json";
import Adapter from "enzyme-adapter-react-16";
import AppCounter from "./AppCounter";

// Enzyme.configure({ adapter: new Adapter() })

// **************** Test App

test("renders learn react link", () => {
  render(<App />);
  const linkElement = screen.getByText(/learn react/i);
  expect(linkElement).toBeInTheDocument();
});

it("renders correctly enzyme", () => {
  const wrapper = shallow(<App />);

  expect(toJson(wrapper)).toMatchSnapshot();
});

// **************** Test App2 origin

it("renders correctly react-test-renderer", () => {
  const renderer = new ShallowRenderer();
  renderer.render(<App2 />);
  const result = renderer.getRenderOutput();

  expect(result).toMatchSnapshot();
});

it("renders correctly enzyme", () => {
  const wrapper = shallow(<App2 />);

  expect(toJson(wrapper)).toMatchSnapshot();
});

test("renders name", () => {
  const renderer = render(<App2 />);
  const linkElement = screen.getByText(/Faisal Akram/i);
  const result = "";
  expect(linkElement).toBeInTheDocument();
});

// **************** Test App2 A

test("renders name", () => {
  render(<App2 />);
  const linkElement = screen.getByText(/Faisal Akram/i);
  expect(linkElement).toBeInTheDocument();
});

// **************** Test Counter

// incorrect function assignment in the onClick method
// will still pass the tests.

test("the increment method increments count", () => {
  const wrapper = shallow(<Counter />);

  expect(wrapper.instance().state.count).toBe(0);

  // wrapper.find('button.counter-button').simulate('click')
  // wrapper.setState({count: 1})
  wrapper.instance().increment();
  expect(wrapper.instance().state.count).toBe(1);
});

// **************** Test Counter 2

afterEach(cleanup);

it("Text in state is changed when button clicked", () => {
  const { getByText } = render(<TestHook />);

  expect(getByText(/Initial/i).textContent).toBe("Initial State");

  fireEvent.click(getByText("State Change Button"));

  expect(getByText(/Initial/i).textContent).toBe("Initial State Changed");
});

it("button click changes props", () => {
  const { getByText } = render(
    <AppCounter>
      <TestHook />
    </AppCounter>
  );

  expect(getByText(/Moe/i).textContent).toBe("Moe");

  fireEvent.click(getByText("Change Name"));

  expect(getByText(/Steve/i).textContent).toBe("Steve");
});

// **************** Test Counter Button increment

describe("test for app", () => {
  it("number gets increased with click", async () => {
    const { getByText } = render(<AppBtnIncrement />);
    expect(getByText("0"));
    fireEvent.click(getByText("Button"));
    await waitFor(() => expect(getByText("1")));
  });
});
