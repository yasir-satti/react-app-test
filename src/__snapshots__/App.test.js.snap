// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly enzyme 1`] = `
<Welcome
  name="Faisal Arkan"
/>
`;
